classfields: eval early errors

spectranaut opened the pull request:
@leo take a look whenever :)

leo commented:
@spectranaut Can you folks please start using the right usernames 😄

spectranaut commented:
oooppss @leobalter XD

leobalter commented on src/class-fields/initializer-eval-super/cls-expr-fields-indirect-eval.template:
```diff
@@ -0,0 +1,21 @@
+  x = (0, eval)('executed = true; /*{ initializer }*/;';
+}
+
+assert.throws(/*{ execution error }*/, function() {
```
interesting, without generating the files I can't tell if the script handles multiple words

rwaldron commented:
> Can you folks please start using the right usernames

😂

rwaldron commented on src/class-fields/initializer-eval/cls-expr-fields-indirect-eval.template:
```diff
@@ -0,0 +1,19 @@
+
+var executed = false;
+C = class {
+  x = (0, eval)('executed = true; /*{ initializer }*/;';
```
Missing a closing `)`

rwaldron commented on src/class-fields/initializer-eval/cls-expr-fields-eval.template:
```diff
@@ -0,0 +1,19 @@
+
+var executed = false;
+C = class {
+  x = eval('executed = true; /*{ initializer }*/;';
```
Missing a closing `)`

rwaldron commented on src/class-fields/initializer-eval/cls-decl-fields-indirect-eval.template:
```diff
@@ -0,0 +1,19 @@
+
+var executed = false;
+class C = {
+  x = (0, eval)('executed = true; /*{ initializer }*/;';
```
Missing a closing `)`


...Ok, I'm not going to be annoying and mark all of these—please fix each file :)

leobalter commented on src/class-fields/eval-err-contains-arguments.case:
```diff
@@ -0,0 +1,29 @@
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+desc: `arguments` in StatementList of eval
```
the initial ` triggers a failure on the yaml parser. This applies to the other cases as well.

leobalter commented on src/class-fields/eval-err-contains-superproperty.case:
```diff
@@ -0,0 +1,28 @@
+---*/
+
+//- initializer
+super().x
```
I believe this should be `super.x` to match a SuperProperty:

```
SuperProperty:
  super [ Expression ]
  super . IdentifierName
```

And this case may also be duplicated with `super['x']`

leobalter commented on src/class-fields/initializer-eval-super/cls-decl-fields-indirect-eval.template:
```diff
@@ -0,0 +1,21 @@
+  x = (0, eval)('executed = true; /*{ initializer }*/;';
+}
+
+assert.throws(/*{ execution error }*/, function() {
```
the reutilization of the templates using execution and early errors fields is great! +1

leobalter replied:
the templates still remain with the "spaced" names.

spectranaut commented:
ok addressed all comments and added generated files, should be good to go @rwaldron @leobalter 

(btw, leo, the space in the template keywords definitely did not work XD removed them)

leobalter commented on src/class-fields/initializer-eval-super/cls-decl-fields-eval.template:
```diff
@@ -0,0 +1,21 @@
+  x = eval('executed = true; /*{ initializer }*/;';
+}
+
+assert.throws(/* early error */, function() {
```
oops

leobalter commented on src/class-fields/initializer-eval/cls-decl-fields-eval.template:
```diff
@@ -0,0 +1,19 @@
+  x = eval('executed = true; /*{ initializer }*/;');
+}
+
+assert.throws(/* earlyerror */, function() {
```
not all of them, thou :)

spectranaut replied:
ohhhnoooo... what's happening?

spectranaut replied:
oh oops I see, forgot "{ }" XD

spectranaut commented:
ok @leo fixed those last few mistakes!

leo commented:
GOD 😄 username
